Only treat a leading "+" as special to open when followed by "<" or
">" (in perlsnap).

diff -Naur perl-5.6.1.orig/doio.c perl-5.6.1/doio.c
--- perl-5.6.1.orig/doio.c	Thu Mar 22 16:05:02 2001
+++ perl-5.6.1/doio.c	Thu Jan 10 23:45:40 2002
@@ -196,7 +196,9 @@
 	}
 	mode[0] = mode[1] = mode[2] = mode[3] = '\0';
 	IoTYPE(io) = *type;
-	if (*type == IoTYPE_RDWR && tlen > 1 && type[tlen-1] != IoTYPE_PIPE) { /* scary */
+	if (*type == IoTYPE_RDWR && tlen > 1 &&
+           (*(type+1) == IoTYPE_RDONLY || *(type+1) == IoTYPE_WRONLY) &&
+	   type[tlen-1] != IoTYPE_PIPE) { /* scary */
 	    mode[1] = *type++;
 	    --tlen;
 	    writing = 1;
